src/main/resources/avro/references.avdl

```diff
@@ -27,8 +105,22 @@ record Reference {
   string name;
 
   /**
+  Start and end locations of how this reference fits into the existing graph.
+  May be null only for a primary sequence that is not joined onto any other.
+  Note that the referenceId for start and end may not be the same.
+  */
+  union { null, ReferenceJoinLocation } startJoin, endJoin;
+
+  /**
+  The sequence to insert between startJoin and endJoin. Must be set if sourceURI
+  is null.
+  */
+  union { null, string } sequence = null;
```

Comment:
I think this model for getting the sequence of a `Reference` has some issues. As I understand it, here either we have the sequence embedded in the `Reference` object (which might be many megabytes) or we provide a URI to the source FASTA file. Presumably we provide this URI when the reference sequence is too long to embed within the `Reference` itself? This creates all sorts of problems in my view: do we expect the client to download and parse the FASTA file out-of-band? 

I think a better approach would be to remove both `sequence` and `sourceURI` from this object, and to provide a `GET` method `getSequence(referenceId)`, which returns the sequence as a string. Problems associated with long strings can be handled using HTTP Streaming, rather than complicating the protocol with paging code for the sequences. (Any thoughts here @calbach and @cassiedoll?)